### DISCUSSION TITLE
async nested model pydantic property

### DISCUSSION THREAD

**Discussion by panla** [pydantic/pydantic#2673] [Question]
### question

- fastapi
- tortoise-orm
- pydantic

```python
# models
from tortoise import fields
from tortoise.models import Model

class User(Model):
    cellphone = fields.CharField(max_length=16, null=False, unique=True)
    ...

class Question(Model):
    owner_id = fields.BigIntField(index=True)
    ...

    @property
    async def owner(self):
        ...
        return await User.get_or_none(id=self.id)

# schema
class OwnerBaseField(BaseModel):
    id: int
    cellphone: str

class QuestionBaseField(BaseModel):
    ...
    owner: Optional[OwnerBaseField]

# api
@router.get('/{q_id}', response_model=QuestionBaseField)
async def read_question(q_id):
    question = await Question.get_or_none(id=q_id)
    return question
```

to get async property function in pydantic model

`question.owner = await question.owner_` is not a good idea, no foreignkey

are there some better ideas?

**Comment by samuelcolvin** [comment]
I'm not entirely clear on the question. In general , though:
* putting too much logic in instance methods on pydantic models is not a good idea, it gets very confusing
* async properties are never a good ida
* using properties for thing that require processing or IO is an anti-pattern and leads to much confusion down the line - I assume if I access a property that it's value is trivial to calculate and I can access it lots without any significant overhead

**Reply by panla** [reply]
thanks,
It seems that only  `question.owner = await User.get_or_none(id=question.id)`,
setattr a property, not a async property on a Model instance.
